Add research intro page (#18)
chaincodelabs/research-site: research-intro.md

We emphasize computer science (cryptography, privacy, networking) and economics, although Bitcoin can be viewed through the lens of other scientific disciplines too.

Bitcoin offers opportunities for applied as well as fundamental research.
The former typically designs and evaluates proposed protocol upgrades planned for the next few years.
The latter tackles more long-term and fundamental challenges that Bitcoin faces.

Bitcoin development emphasizes security and stability.
Changes to the protocol pass strict review.
Even though moving academic results towards implementation may be a challenging task,
high-quality scientific contributions to Bitcoin will have a lasting impact.

We hope this page helps you choose your research direction.

The Bitcoin community looks forward to your contributions!

# Introduction

Let us start with introductory materials to get a general overview of the area.

Books:

* Narayanan et al. [Bitcoin and Cryptocurrency Technologies](https://bitcoinbook.cs.princeton.edu/)
* Antonopoulos. [Mastering Bitcoin](https://aantonop.com/books/mastering-bitcoin/)
* Antonopoulos et al. [Mastering the Lightning Network](https://github.com/lnbook/lnbook)

> * Rosenbaum. [Grokking Bitcoin](https://www.manning.com/books/grokking-bitcoin)

Reviewer: The list of books could be very long. What is the criteria for inclusion?

Author: It's largely subjective. In particular, the Princeton one is the first academic introductory book on the subject, as far as I know, and I like the fact that it's accompanied by a MOOC.

What changes would you suggest?

Introductory and systematization-of-knowledge papers:

* Nakamoto. [Bitcoin: A Peer-to-Peer Electronic Cash System](https://bitcoin.org/bitcoin.pdf)
* Narayanan and Clark. [Bitcoin's Academic Pedigree](https://queue.acm.org/detail.cfm?id=3136559)
* Bonneau et al. [SoK: Research Perspectives and Challenges for Bitcoin and Cryptocurrencies](https://www.ieee-security.org/TC/SP2015/papers-archived/6949a104.pdf)
* Tschorsch and Scheuermann. [Bitcoin and Beyond: A Technical Survey on Decentralized Digital Currencies](https://eprint.iacr.org/2015/464)
* Gudgeon et al. [SoK: Layer-Two Blockchain Protocols](https://eprint.iacr.org/2019/360)

Collections of materials compiled at Chaincode Labs with a stronger focus on development:

> * [Chaincode's Bitcoin Curriculum](https://github.com/chaincodelabs/bitcoin-curriculum)

Reviewer: These are pretty outdated. Not sure they should be included.

* [Chaincode's Lightning Curriculum](https://github.com/chaincodelabs/lightning-curriculum)

# Research Areas in Bitcoin

> This section lists some of the prior work and research challenges related to the core Bitcoin protocol.

Reviewer 2: I think "core Bitcoin protocol" is a bit misleading for all the listed papers. It's not just about the Bitcoin P2P protocol, nor just about Bitcoin's consensus rules.

For example, all the cryptography papers are related to multisignatures and threshold signatures - a concept that doesn't even exist at the consensus or P2P layer in Bitcoin; it's something that certain wallets choose to implement, or something that's a basis for other protocols. Some of the economics papers discuss protocol changes that are unlikely to ever be adopted (like the red balloons paper).

(I don't object to grouping all non-Lightning stuff together, but I'm not sure what to call it, as it's pretty diverse)

Author: Summarizing our offline conversation: I'll re-frame signature-related section as, well, "signature schemes", and I'll also add a comment on how the research space is multi-dimensional, and we present just one of many possible ways to group together related papers.

## Cryptography

The main cryptographic primitives Bitcoin uses are digital signatures and hash functions.
Most recently, research efforts have been put into implementing secure cryptographic schemes based on a new type of signatures (Schnorr signatures) that Bitcoin supports since 2021.

> Relevant work on Bitcoin-related cryptography includes:

Reviewer 2: Perhaps it's worth adding the MuSig papers (there are 3, MuSig: https://eprint.iacr.org/2018/068, MuSig-DN: https://eprint.iacr.org/2020/1057, MuSig2: https://eprint.iacr.org/2020/1261). I'd say MuSig2 is the most practically relevant one, and MuSig-DN the least, but they're both building on top of ideas in MuSig.

* Gennaro et al. [Threshold-optimal DSA/ECDSA signatures and an application to Bitcoin wallet security](https://eprint.iacr.org/2016/013)
* Komlo and Goldberg. [FROST: Flexible Round-Optimized Schnorr Threshold Signatures](https://eprint.iacr.org/2020/852)
* Ruffing et al. [ROAST: Robust Asynchronous Schnorr Threshold Signatures](https://eprint.iacr.org/2022/550)

## P2P Network

> Bitcoin nodes broadcast transactions in the P2P network.

Reviewer 3: Maybe this sentence could be a bit more general to cover more topics from the papers listed below - seems a bit too much centered on tx relay.

Author: "Bitcoin uses a P2P network to broadcast transactions, blocks, and other data." - would this be better?

To allow public verifiability and establishing a level playing field for miners, information propagation should happen efficiently even in low-bandwidth settings, while protecting from network-based privacy attacks.

Prior work on P2P networking in the context of Bitcoin includes:

* Apostolaki et al. [Hijacking Bitcoin: Routing Attacks on Cryptocurrencies](https://ieeexplore.ieee.org/document/7958588)
* Decker and Wattenhofer. [Information propagation in the Bitcoin network](https://ieeexplore.ieee.org/abstract/document/6688704)
* Dotan et al. [Survey on Cryptocurrency Networking: Context, State-of-the-Art, Challenges](https://arxiv.org/abs/2008.08412)
* Fanti et al. [Dandelion++: Lightweight Cryptocurrency Networking with Formal Anonymity Guarantees](https://arxiv.org/abs/1805.11060)

> * Heilman and Kendler. [Eclipse Attacks on Bitcoin’s Peer-to-Peer Network](https://www.usenix.org/conference/usenixsecurity15/technical-sessions/presentation/heilman)

Reviewer 3: Heilman et al.

* Naumenko et al. [Erlay: Efficient Transaction Relay for Bitcoin](https://dl.acm.org/doi/10.1145/3319535.3354237)
* Neudecker and Hartenstein. [Network Layer Aspects of Permissionless Blockchains](https://ieeexplore.ieee.org/document/8456488)

> 

Reviewer: I'd suggest [erebus](https://erebus-attack.comp.nus.edu.sg/) as well

## Privacy

Privacy is a vital property of Bitcoin.
Still, multiple privacy attacks have been described in the literature.
The general research question is achieving a high level of privacy and fungibility the the P2P setting with the transaction data being public for verifiability.

One class of privacy attacks implies analyzing P2P network traffic.
Prior work in this area includes:

* Biryukov and Pustogarov. [Bitcoin over Tor isn't a good idea](https://arxiv.org/abs/1410.6079)
* Biryukov et al. [Deanonymisation of clients in Bitcoin P2P network](https://arxiv.org/abs/1405.7418)
* Fanti and Viswanath. [Anonymity Properties of the Bitcoin P2P Network](https://arxiv.org/abs/1703.08761)

> * Koshy et al. [An Analysis of Anonymity in Bitcoin Using P2P Network Traffic](https://www.ifca.ai/fc14/papers/fc14_submission_71.pdf)

Reviewer 2: Maybe worth including TxProbe: https://arxiv.org/abs/1812.00942

* Garay et al. [The Bitcoin Backbone Protocol: Analysis and Applications](https://eprint.iacr.org/2014/765)
* Kroll et al. [The Economics of Bitcoin Mining, or Bitcoin in the Presence of Adversaries](https://asset-pdf.scinapse.io/prod/2188530018/2188530018.pdf)

## Economics

Bitcoin can also be viewed from the economics perspective.
Papers on economic analysis of Bitcoin include:

* Budish. [The Economic Limits of Bitcoin and Anonymous, Decentralized Trust on the Blockchain](https://papers.ssrn.com/sol3/papers.cfm?abstract_id=4148014)
* Böhme et al. [Bitcoin: Economics, Technology, and Governance](https://www.aeaweb.org/articles?id=10.1257/jep.29.2.213)
* Huberman et al. [Monopoly without a Monopolist: An Economic Analysis of the Bitcoin Payment System](https://papers.ssrn.com/sol3/papers.cfm?abstract_id=3025604)
* Kayal and Rohilla. [Bitcoin in the economics and finance literature: a survey](https://link.springer.com/article/10.1007/s43546-021-00090-5)

> # Research Areas in Lightning

Reviewer 3: Could mention the original lighting paper by Poon/Dryja somewhere in the document.

Author: I thought about this, but was hesitant to include it because a) it's not an easy read; b) it doesn't reflect the current state of the protocol. From personal experience, I remember spending quite some effort to read it carefully, only to  discover later on that the actual specs don't have "breach remedy transactions" in the same shape as the paper introduces.

Perhaps it should be included for historical reasons (like the Nakamoto whitepaper) with a comment that the protocol has diverged from it considerably?

Reviewer 3: Yes, seems like a good idea to treat these two papers similarly.

* Tschorsch and Scheuermann. [Bitcoin and Beyond: A Technical Survey on Decentralized Digital Currencies](https://eprint.iacr.org/2015/464)
* Gudgeon et al. [SoK: Layer-Two Blockchain Protocols](https://eprint.iacr.org/2019/360)

Multiple books have been written on the subject, such as:

* Narayanan et al. [Bitcoin and Cryptocurrency Technologies](https://bitcoinbook.cs.princeton.edu/)
* Antonopoulos. [Mastering Bitcoin](https://aantonop.com/books/mastering-bitcoin/)
* Antonopoulos et al. [Mastering the Lightning Network](https://github.com/lnbook/lnbook)
* Rosenbaum. [Grokking Bitcoin](https://www.manning.com/books/grokking-bitcoin)

While it's not necessarily required to read those books cover to cover to start with Bitcoin research, they can be helpful as points of reference on Bitcoin's technical details.

# A path from research to development in Bitcoin

Bitcoin is guided by a [design philosophy](https://github.com/bitcoin-dev-philosophy/btcphilosophy) that puts a strong emphasis on decentralization and trustlessness.
As a consequence, Bitcoin is difficult to change.

> Proposed modifications that break backwards compatibility (i.e., introduce a _hard fork_) are highly unlikely to be considered.

Reviewer 2: I think this paragraph conflates consensus rule changes with protocol changes. You talk about softforks and hardforks here, but the three examples below are all about P2P protocol/behavior changes, and not related to any softfork or hardfork.

Author: Fair point. I guess what I'm trying to convey here are two things:

1. changes in Bitcoin are slow (and for a good reason), so you (the researcher) shouldn't expect your proposal to be implemented within months, even if it comes from an academically solid paper;
2. even under these constraints, there has been cases of academic work that lead to actual code changes.

Have there been examples of "academically"-inspired consensus-level changes though?

---
layout: page
title: Introduction to Bitcoin Research
order: 4
---

Hello, Researcher!

We appreciate your interest in Bitcoin. 

Bitcoin is a fascinating and multi-faceted topic.
It offers challenging questions across a variety of fields of study.
This page provides an overview of Bitcoin-related research.

> It focuses on computer science (cryptography, privacy, networking) and economics, although Bitcoin can be viewed through the lens of other scientific disciplines too.

Reviewer 4: ```suggestion
It focuses on computer science (cryptography, privacy, networking) and economics, although Bitcoin can be viewed through the lens of other scientific disciplines, too.
```

* Narayanan and Clark. [Bitcoin's Academic Pedigree](https://queue.acm.org/detail.cfm?id=3136559)
* Bonneau et al. [SoK: Research Perspectives and Challenges for Bitcoin and Cryptocurrencies](https://www.ieee-security.org/TC/SP2015/papers-archived/6949a104.pdf)
* Tschorsch and Scheuermann. [Bitcoin and Beyond: A Technical Survey on Decentralized Digital Currencies](https://eprint.iacr.org/2015/464)
* Gudgeon et al. [SoK: Layer-Two Blockchain Protocols](https://eprint.iacr.org/2019/360)

Multiple books have been written on the subject, such as:

* Narayanan et al. [Bitcoin and Cryptocurrency Technologies](https://bitcoinbook.cs.princeton.edu/)
* Antonopoulos. [Mastering Bitcoin](https://aantonop.com/books/mastering-bitcoin/)
* Antonopoulos et al. [Mastering the Lightning Network](https://github.com/lnbook/lnbook)
* Rosenbaum. [Grokking Bitcoin](https://www.manning.com/books/grokking-bitcoin)

While it's not necessarily required to read those books cover to cover to start with Bitcoin research, they can be helpful as points of reference on Bitcoin's technical details.

# A path from research to development in Bitcoin

> Bitcoin is guided by a [design philosophy](https://github.com/bitcoin-dev-philosophy/btcphilosophy) that strongly emphasizes decentralization and trustlessness.

Reviewer 4: Perhaps:
```suggestion
Bitcoin is guided by a [design philosophy](https://github.com/bitcoin-dev-philosophy/btcphilosophy) that strongly emphasizes decentralization and trust minimization.
```

Relevant work includes:

* Nick et al. [MuSig2: Simple Two-Round Schnorr Multi-Signatures](https://eprint.iacr.org/2020/1261) proposes a practical multi-signature scheme for Schnorr signatures (related prior work includes [MuSig](https://eprint.iacr.org/2018/068) and [MuSig-DN](https://eprint.iacr.org/2020/1057)).
* Komlo and Goldberg. [FROST: Flexible Round-Optimized Schnorr Threshold Signatures](https://eprint.iacr.org/2020/852) proposes a threshold signature scheme for Schnorr signatures.
* Ruffing et al. [ROAST: Robust Asynchronous Schnorr Threshold Signatures](https://eprint.iacr.org/2022/550) adapts signature schemes like FROST to an asynchronous setting.

## P2P Network

Bitcoin uses a P2P network to broadcast transactions, blocks, and related data.
Inspired by earlier data sharing P2P protocols, Bitcoin operates under unique design constraints.
To allow public verifiability and establishing a level playing field, the network should ensure efficient information propagation even in low-bandwidth settings.
At the same time, it should defend against denial-of-service and privacy attacks.

> For a review of eclipse attacks and countermeasures against them, see [this document](https://github.com/bitcoin-core/bitcoin-devwiki/wiki/Addrman-and-eclipse-attacks) by Martin Zumsande.

Reviewer 3: It's a collaborative wiki document, and ~95% of it has been written by @adamjonas, I'm just the most recent person to edit it.

* Garay et al. [The Bitcoin Backbone Protocol: Analysis and Applications](https://eprint.iacr.org/2014/765) formally describe the Bitcoin protocol and analyze it in the context of Byzantine agreement.
* Pass et al. [Analysis of the Blockchain Protocol in Asynchronous Networks](https://eprint.iacr.org/2016/454) analyzes Nakamoto consensus protocol in the asynchronous setting.
* Guo and Ren. [Bitcoin's Latency--Security Analysis Made Simple](https://arxiv.org/abs/2203.06357) develop the upper and lower bounds on the security of Nakamoto consensus.

## Economics

Bitcoin can also be viewed from the economics perspective.

Research questions in Bitcoin as viewed through the economic lens may include:

* **Bitcoin price analysis.** What influences the price of Bitcoin? See e.g. [How Futures Trading Changed Bitcoin Prices](https://www.frbsf.org/economic-research/wp-content/uploads/sites/4/el2018-12.pdf) by Hale et al.

> * **Optional transaction fee policies.** How is Bitcoin's block space market organized? What do fee rates depend upon? How should wallets construct transactions from their available unspent transaction outputs (see [Coin selection](https://bitcoinops.org/en/topics/coin-selection/) and [An Evaluation of Coin SelectionStrategies](https://murch.one/erhardt2016coinselection.pdf) by Mark Erhardt)?

Reviewer 4: ```suggestion
* **Optional transaction fee policies.** How is Bitcoin's block space market organized? What do fee rates depend upon? How should wallets construct transactions from their available unspent transaction outputs (see [Coin selection](https://bitcoinops.org/en/topics/coin-selection/) and [An Evaluation of Coin Selection Strategies](https://murch.one/erhardt2016coinselection.pdf) by Mark Erhardt)?
```
